gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/support/PubSubConfigurationFactory.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.pubsub.support;
+
+import com.google.api.gax.core.ExecutorProvider;
+import com.google.api.gax.core.FixedExecutorProvider;
+import com.google.api.gax.grpc.InstantiatingGrpcChannelProvider;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import com.google.cloud.pubsub.v1.Publisher;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.gcp.GoogleCloudConfiguration;
+import io.micronaut.gcp.UserAgentHeaderProvider;
+import io.micronaut.gcp.pubsub.configuration.PubSubConfigurationProperties;
+import org.threeten.bp.Duration;
+
+import javax.inject.Singleton;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * Factory class to create default settings for PubSub Publisher and subscriber beans.
+ *
+ * @author Vinicius Carvalho
+ * @since 2.0.0
+ *
+ */
+@Factory
+@Requires(classes = Publisher.class)
+public class PubSubConfigurationFactory {
+
+    private final PubSubConfigurationProperties pubSubConfigurationProperties;
+
+    private final GoogleCloudConfiguration googleCloudConfiguration;
+
+    public PubSubConfigurationFactory(PubSubConfigurationProperties pubSubConfigurationProperties, GoogleCloudConfiguration googleCloudConfiguration) {
+        this.pubSubConfigurationProperties = pubSubConfigurationProperties;
+        this.googleCloudConfiguration = googleCloudConfiguration;
+    }
+
+    /**
+     *
+     * @return default {@link ExecutorProvider}
+     */
+    @Singleton
+    public ExecutorProvider publisherExecutorProvider() {
```

Comment:
We have ways to create thread pools via `micronaut.executors` configuration. This should support a user defined executor `gcp.pubsub.subscriber-executor: abc`, and default to using `@Named("scheduled") ExecutorService executorService`.

Reply:
@jameskleeh So I have a couple of pending comments only to fix, this being one of them. I could not find a lot of info on the micronaut.executors besides some source code on the Scheduling system. 
I've just changed the publisher properties to use `@EachProperty` which was my intention since day one. Makes more sense. So let me ask you this, suppose each `PubSubPublisherProperties` has its own `subscriber-executor` name, how do I go about creating that executor, just manually or do we have a factory for such things?
Also, I'm assuming I would inject the default `@Named("scheduled")` onto the `PublisherFactory` and if no user defined is present use that one right?

Reply:
@viniciusccarvalho Thats right. For example the rabbitmq allows the executor to be set via the annotation: https://github.com/micronaut-projects/micronaut-rabbitmq/blob/master/rabbitmq/src/main/java/io/micronaut/rabbitmq/intercept/RabbitMQConsumerAdvice.java#L191. It was done that way because there isn't per consumer configuration. I think that may be the case here so it may be worth doing the same thing (having it in the annotation instead of config)